### PR TITLE
Fix Filter Views for CSV Reg Report

### LINKED ISSUE #3899: Fix searching within the registration list removing filters
Reported here: https://eventespresso.com/topic/csv-exports-ee4/

If you apply a filter within the registration list table and then use search to narrow down your selection... when you export the results none of the filters (including the search) apply to the export.

**To Reproduce**
I use an event with 2 different ticket types to test this.

Have registrations on both ticket types and have some that are pending payment (or just something other than Approved).

Before: https://monosnap.com/file/ujmChfmG47ehfzNPLGZ7X8Lx6lCaNR

Export works fine.

Set the Reg status to Approved, filter: https://monosnap.com/file/nhBQKubaib0RLRjbcRRzRfxYW8cBSv

My export now contains 7 registrations, as expected.

Now add something to the search field, in my example I'm filtering to another specific ticket name: https://monosnap.com/file/9z4TfkaFOEa6u4CPBzPoMHzoTHJ3B5

Export now contains ALL registrations, including the pending payment reg that was filtered previously, so in this example I'm getting 8 regs exported, should be 2 as above.

(Priority high due to previous issues with exports)

### LINKED ISSUE #3901: Fix views no longer being applied to registration reports
@joaovictorangeline mentioned that the 'Views' on the registrations table did not apply to the registration report during testing of #3875 , specifically [HERE](https://github.com/eventespresso/event-espresso-core/pull/3875#issuecomment-1149378953)

Afaik those views did not apply to the report, however, I've been testing the difference between v4.10.29 and the current version with regards to the CSV reports and those views do indeed apply in v29 when using the 'Filtered CSV Report' button (my apologies to @joaovictorangeline )

----

To reproduce simply go to Event Espresso -> Registrations.

Use say the 'This Month' view: https://monosnap.com/file/MHeGrOL1UCpZe969xlR2470qWl7jj4

You can see I get 8 registrations returned and that's what is exported when I click the 'Filtered CSV Report'

In master, I get the same registration list table view but on export, it exports all of my sites registrations

### PR DESCRIPTION
**Pull request by tn3rb**
This PR fixes #3901 && fixes #3899

### CODE REVIEW

File: core/admin/EE_Admin_List_Table.core.php [hunk 1]
```diff
@@ -735,7 +739,7 @@ public function get_hidden_columns()
      */
     public function single_row_columns($item)
     {
-        [$columns, $hidden, $sortable, $primary] = $this->get_column_info();
+        list($columns, $hidden, $sortable, $primary) = $this->get_column_info();
```
Comment (hoseinrafiei) [inline core/admin/EE_Admin_List_Table.core.php:742]: Is something wrong with the [] syntax? Curios to know why do you reverted to the classic style? 🤔

File: core/admin/EE_Admin_List_Table.core.php [hunk 2]
```diff
@@ -806,8 +810,8 @@ public function extra_tablenav($which)
         } else {
             echo '<div class="list-table-bottom-buttons alignleft actions">';
             foreach ($this->_bottom_buttons as $type => $action) {
-                $route         = $action['route'] ?? '';
-                $extra_request = $action['extra_request'] ?? '';
+                $route         = isset($action['route']) ? $action['route'] : '';
+                $extra_request = isset($action['extra_request']) ? $action['extra_request'] : '';
```
Comment (hoseinrafiei) [inline core/admin/EE_Admin_List_Table.core.php:814]: They are equivalent and null coalescing operator is cleaner and more readable. Any reason to switch to classic syntax?

**Review by hoseinrafiei — APPROVED** [review]

File: core/admin/EE_Admin_List_Table.core.php (hunk 1 above)
Comment (tn3rb) [reply core/admin/EE_Admin_List_Table.core.php:742]: my IDE converts some of these things automatically, but this branch will be getting merged into master which still allows PHP 5.6, but PHP 5.6 does not allow array destructuring nor null coalescing operators

File: core/admin/EE_Admin_List_Table.core.php (hunk 2 above)
Comment (tn3rb) [reply core/admin/EE_Admin_List_Table.core.php:814]: i agree but master branch still allows PHP 5.6

**Comment by Pebblo** [comment]
@tn3rb I'm still getting the same issues #3899 & #3901 on this branch.

A quick test is to use the 'Today' view, I have 1 registration for today but I'm getting a full export

Using search on a filtered report still returns all of the registrations for the current event.

**Review by knazart — APPROVED** [review]

**Comment by tn3rb** [comment]
sry about that Tony

I had tested views while ALSO using the filters, but had not tested using views on their own. 
The way i had the code was only applying the views if you were also using other filters, but that should be fixed now

**Comment by garthkoyle** [comment]
What I found exporting registration data:

* [x] FILTERS ONLY works
* [x] SEARCH + FILTERS outputs all data 💥 when it should be less data (usually a subset of the filters only)
* [ ] SEARCH ONLY exports all data.

**Comment by garthkoyle** [comment]
okay, so now this generates the right csv: 

- no filters
- filters
- search
- filters + search

**Pull request merged**